domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/calendardates/CalendarDate.java

```diff
@@ -0,0 +1,156 @@
+    @NotNull
+    private final LocalDateTime date;
+    @NotNull
+    final ExceptionType exceptionType;
+
+    /**
+     * @param serviceId     identifies a set of dates when a service exception occurs for one or more routes
+     * @param date          date when service exception occurs
+     * @param exceptionType indicates whether service is available on the date specified in the date field
+     */
+    private CalendarDate(@NotNull String serviceId,
+                         @NotNull LocalDateTime date,
+                         @NotNull ExceptionType exceptionType) {
+        this.serviceId = serviceId;
+        this.date = date;
+        this.exceptionType = exceptionType;
+    }
+
+    @NotNull
+    public String getServiceId() {
+        return serviceId;
+    }
+
+    @NotNull
+    public LocalDateTime getDate() {
+        return date;
+    }
+
+    @NotNull
+    public ExceptionType getExceptionType() {
+        return exceptionType;
+    }
+
+    /**
+     * Builder class to create {@link CalendarDate} objects. Allows an unordered definition of the different attributes
+     * of {@link CalendarDate}.
+     */
+    public static class CalendarDateBuilder {
+        private String serviceId;
+        private LocalDateTime date;
+        private ExceptionType exceptionType;
+        private Integer originalExceptionTypeInteger;
+        private final List<Notice> noticeCollection = new ArrayList<>();
+
+        /**
+         * Sets field serviceId value and returns this
+         *
+         * @param serviceId identifies a set of dates when a service exception occurs for one or more routes
+         * @return builder for future object creation
+         */
+        public CalendarDateBuilder serviceId(@NotNull final String serviceId) {
+            this.serviceId = serviceId;
+            return this;
+        }
+
+        /**
+         * Sets field date value and returns this
+         *
+         * @param date date when service exception occurs
+         * @return builder for future object creation
+         */
+        public CalendarDateBuilder date(@NotNull final LocalDateTime date) {
+            this.date = date;
+            return this;
+        }
+
+        /**
+         * Sets field exceptionType value and returns this
+         *
+         * @param exceptionType indicates whether service is available on the date specified in the date field
+         * @return builder for future object creation
+         */
+        public CalendarDateBuilder exceptionType(@NotNull final Integer exceptionType) {
+            this.exceptionType = ExceptionType.fromInt(exceptionType);
+            this.originalExceptionTypeInteger = exceptionType;
+            return this;
+        }
+
+        /**
+         * Returns an entity representing a row from calendar_dates.txt if the requirements from the official GTFS
+         * specification are met. Otherwise, method returns an entity representing a list of notices.
```

Comment:
There's ambiguity in the use of the term 'entity' at the end of the sentence.
Suggestion: Otherwise return a collection of notices describing the issue